src/github3/users.py

```diff
@@ -12,6 +12,103 @@
 from .events import Event
 
 
+class GPGKey(models.GitHubCore):
+    """The object representing a user's GPG key.
+
+    .. versionadded:: 1.2.0
+
+    Please see GitHub's `GPG Key Documentation` for more information.
+
+    .. _GPG Key Documentation:
+        https://developer.github.com/v3/users/gpg_keys/
+
+    .. attribute:: can_certify
+
+        TODO
+
+    .. attribute:: can_encrypt_comms
+
+        TODO
+
+    .. attribute:: can_encrypt_storage
+
+        TODO
+
+    .. attribute:: can_sign
+
+        TODO
+
+    .. attribute:: created_at
+
+        A :class:`~datetime.datetime` representing the date and time when
+        this GPG key was created.
+
+    .. attribute:: emails
+
+        TODO.
+
+    .. attribute:: expires_at
+
+        A :class:`~datetime.datetime` representing the date and time when
+        this GPG key will expire.
+
+    .. attribute:: id
+
+        The unique identifier of this GPG key.
+
+    .. attribute:: key_id
+
+        TODO
+
+    .. attribute:: primary_key_id
+
+        TODO
+
+    .. attribute:: public_key
+
+        TODO
+
+    .. attribute:: subkeys
+
+        TODO
+    """
+
+    def _update_attributes(self, key):
+        self.can_certify = key['can_certify']
+        self.can_encrypt_comms = key['can_encrypt_comms']
+        self.can_encrypt_storage = key['can_encrypt_storage']
+        self.can_sign = key['can_sign']
+        self.created_at = self._strptime(key['created_at'])
+        self.emails = [Email(email, self) for email in key['emails']]
+        try:
+            self.expires_at = self._strptime(key['expires_at'])
+        except ValueError:
```

Comment:
Is this not valid in practice?

Reply:
I thought that `_strptime()` handled that case automagically. If not, it should.

Reply:
Whoops, you are right! See: https://github.com/sigmavirus24/github3.py/blob/aba7b2381a38f3c58749c3e6b71485275b4e69ed/src/github3/models.py#L87-L100

I was probably thinking of `datetime`'s `strptime`. Will fix this too!